### DISCUSSION TITLE
Build failing due to serverless function maximum size limit

### DISCUSSION THREAD

**Discussion by bddicken** [vercel/vercel#11627] [General]
### Description

I am working on adding some new pages to a site that we deploy using Vercel. The site deploys just fine on our `main` branch. I created a branch of our site, added a few components and markdown files, made a few code edits, and then push to Vercel. However, the Vercel build fails with:

```
Error: The Serverless Function "sync/resources" is 1.08gb which exceeds the maximum size limit of 300mb. Learn More: https://vercel.link/serverless-function-size
```

Earlier up in the logs I see:

```
...

Warning: Max serverless function size of 250 MB uncompressed reached
--
11:21:51.528 | Serverless Function's page: learn/articles/[slug].js
11:21:51.538 | Large Dependencies            Uncompressed size
11:21:51.538 | .next/cache/webpack                     1.56 GB
11:21:51.539 | .git/objects/pack                     406.11 MB
11:21:51.539 | public/assets/blog                    276.14 MB
11:21:51.539 | public/assets/docs                     61.27 MB
...
```

However, I'm not sure why these large dependencies are being included here but not on `main`. I did not add any additional packages or dependencies on my working branch. When I run `vercel build` locally it builds successfully, but I do get this warning, which is probably related:

```
WARNING: You should not upload the `.next` directory.
```

I did not make any changes explicitly ask it to include `.next`. Anyone else experienced this? Pointers in the right direction?

**Comment by lforst** [comment]
We are facing the same issue on the Sentry docs when bumping Next.js from 14.0.2 to 14.2.4. I assume there is something depending on the webpack cache and git in the nextjs code itself and nft picks it up.

**Comment by AdrianMachado** [comment]
Also hitting this error migrating from 14.0.2 to 14.2.4

**Comment by Godrules500** [comment]
Running into the same issue.
